lib/Frontend/ParseableInterfaceModuleLoader.cpp

```diff
@@ -825,7 +828,7 @@ class ParseableInterfaceModuleLoaderImpl {
       // If it's a forwarding module, load the YAML file from disk and check
       // if it's up-to-date.
       if (isForwardingModule) {
-        auto modOrErr = ForwardingModule::load(buf, fs);
+        auto modOrErr = ForwardingModule::load(buf);
         if (modOrErr && forwardingModuleIsUpToDate(*modOrErr, deps))
           return DiscoveredModule::forwarded(modOrErr->underlyingModulePath);
```

Comment:
What happens if the underlying module is gone or corrupted? :-/ Is *that* caught later?

Reply:
That check is done in `forwardingModuleIsUpToDate`

```c++
// First, make sure the underlying module path exists and is valid.
    auto modBuf = getSerializedASTBuffer(fm, fwd.underlyingModulePath);
    if (!modBuf || !serializedASTLooksValid(*modBuf))
      return false;
```

Reply:
And there's a test for it

Reply:
Aha! Thanks.